app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -144,6 +144,12 @@ def perform_delete_in_cloud
       logger.warn("VMT with #{id_at_site} does not exist - continuing")
     end
 
+    # Returns the hourly cost for this template assuming a given VM flavor
+    def get_hourly_cost_for(flavor)
+      incarnation = flavor.flavor_os_families.find_by(os_family: appliance_type.os_family)
+      incarnation && incarnation.hourly_cost
+    end
```

Comment:
@mkasztelnik This won't address the issue at hand (a template may represent an OS family which does not exist on its current compute site - quite likely, in fact, if a Windows template is copied to a Linux-only site). Besides, I can't imagine a situation where an `OSFamily` object is destroyed - except perhaps when added in error.